tests/integration/test_clip.py

```diff
@@ -0,0 +1,180 @@
+from utils.generators import mk_seq_array
+
+import cunumeric as num
+
+# cunumeric.clip(array, a_min, a_max, out=None, **kwargs)
+
+
+@pytest.mark.xfail
+def test_none_array():
+    expected_exc = TypeError
+    with pytest.raises(expected_exc):
+        np.clip(None, a_min=0, a_max=0)
+    with pytest.raises(expected_exc):
+        # cunumeric raises
+        # AttributeError: 'NoneType' object has no attribute 'clip'
+        num.clip(None, a_min=0, a_max=0)
+
+
+def test_empty_array():
+    res_np = np.clip([0], a_min=0, a_max=0)
+    res_num = num.clip([0], a_min=0, a_max=0)
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+def test_value_none():
+    array = np.arange(0, 10)
+    expected_exc = ValueError
+    with pytest.raises(expected_exc):
+        # Numpy raises:
+        # ValueError: One of max or min must be given
+        np.clip(array, a_min=None, a_max=None)
+    with pytest.raises(expected_exc):
+        # cunumeric raises:
+        # TypeError: int() argument must be a string, a bytes-like object
+        # or a real number, not 'NoneType'
+        num.clip(array, a_min=None, a_max=None)
+
+
+@pytest.mark.xfail
+def test_amin_amax():
+    array = np.arange(0, 10)
+    res_np = np.clip(array, a_min=9, a_max=5)
+    res_num = num.clip(array, a_min=9, a_max=5)
+    # the result is different
+    # array = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    # res_np = array([5, 5, 5, 5, 5, 5, 5, 5, 5, 5])
+    # res_num = array([9, 9, 9, 9, 9, 9, 9, 9, 9, 5])
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+@pytest.mark.parametrize("amin", (-1, 0.5, 2.5, 5, 11))
+def test_amin_value(amin):
+    array = np.arange(0, 10)
+    res_np = np.clip(array, a_min=amin, a_max=8.5)
+    res_num = num.clip(array, a_min=amin, a_max=8.5)
+    # res_np is not match res_num
+    # in Numpy, if a_min or a_max is float, all data are marked as float,
+    # while in cunumeric, all datas are int.
+    # for example, amin = 5
+    # array = array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+    # res_np = array([5., 5., 5., 5., 5., 5., 6., 7., 8., 8.5])
+    # res_num = array([5, 5, 5, 5, 5, 5, 6, 7, 8, 8])
+    assert np.array_equal(res_np, res_num)
+
+
+@pytest.mark.xfail
+def test_amin_complex():
+    array = np.arange(0, 10)
+    amin = 5 + 5j
+    res_np = np.clip(array, a_min=amin, a_max=8.5)
+    #  res_np = array([5. +5.j, 5. +5.j, 5. +5.j, 5. +5.j, 5. +5.j,
+    #   5. +5.j, 6. +0.j, 7. +0.j, 8. +0.j, 8.5+0.j])
+    res_num = num.clip(array, a_min=amin, a_max=8.5)
+    # cunumeric raises:
+    # TypeError: int() argument must be a string, a bytes-like object
+    # or a real number, not 'complex'
+    assert np.array_equal(res_np, res_num)
+
+
+def test_value_list():
+    array = np.arange(0, 5)
+    amin = [2, 3, 4, 5, 1]
+    amax = 8
+    res_np = np.clip(array, a_min=amin, a_max=amax)
+    res_num = num.clip(array, a_min=amin, a_max=amax)
+    assert np.array_equal(res_np, res_num)
+
+
+def test_value_list_negative():
```

Comment:
When I first read this I assumed it was to do with negative indices or something. Maybe "error"would have a less ambiguous connotation than "negative" for these test names?

Alternatively, classes can be nice for grouping in cases like these, then the individual test names can be shorter and simpler since they are under a group:
```
class TestClipErrors:

    def test_value_list(self): ...
    def test_out(self): ...

```

Reply:
fix